pinot-core/src/main/java/org/apache/pinot/core/plan/GroupByPlanNode.java

```diff
@@ -59,6 +59,7 @@ public Operator<GroupByResultsBlock> run() {
     assert _queryContext.getGroupByExpressions() != null;
 
     if (_queryContext.hasFilteredAggregations()) {
+      assert _queryContext.getFilteredAggregationFunctions() != null;
```

Comment:
For the context, this is never `null` even if there is no filter on aggregation

Reply:
I've updated `QueryContext` annotation for this field to be `Nonnull` rather than `Nullable` as this has caught me a few times

Reply:
We cannot annotate it as `Nonnull` because it is `Nonnull` only for aggregation queries. Basically `_queryContext.getAggregationFunctions()` and `_queryContext.getFilteredAggregationFunctions()` will both be `null` or `non-null`.